### PR TITLE
port NidecBrushless.java

### PR DESCRIPTION
**Pull request by ariovistus**
hal is throwing a fit, not sure what the deal is

### CODE REVIEW

**Comment by Twinters007** [comment]
Found the issue. 

https://github.com/robotpy/robotpy-wpilib/blob/master/wpilib/wpilib/digitaloutput.py#L1

Someone didn't actually update this file

**Comment by ariovistus** [comment]
I did update digitaloutput here: https://github.com/robotpy/robotpy-wpilib/commit/ac0b85bece08d2133b54bccb670eff04e4a90275#diff-b77bced4d9ad75caee157f3acfc4c45c

Upstream was subsequently updated with a typo fix, which I also pushed

**Comment by Twinters007** [comment]
Yeah, I spent time going through and it looks like it was last year that it didn't get updated

File: wpilib/wpilib/nidecbrushless.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        Gets the channel number associated with the object.
+        :returns: The channel number.
+        :rtype: int
+        """
+        return self.pwm.getChannel()
+
+    def getSmartDashboardType(self):
+        """
+        Live Window code, only does anything if live window is activated.
+        """
```
Comment (auscompgeek) [inline wpilib/wpilib/nidecbrushless.py]: Upstream does not have a corresponding Javadoc for this method.

**Comment by Twinters007** [comment]
Should rely on #331

**Comment by auscompgeek** [comment]
@Twinters007 #331 does not fix the tests for this PR.

**Comment by Twinters007** [comment]
Yeah I guess I forgot why I started messing with those in the first place

**Comment by auscompgeek** [comment]
Yeah, it's purely because hal-sim is broken. I don't have time to comb through it and fix it right now though.

**Comment by Twinters007** [comment]
Yeah I found the root problem and am now trying to figure out what wpilib is actually doing, so that I can replicate. I'm not as good with cpp as I am with java so it's a little confusing

File: wpilib/wpilib/nidecbrushless.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        :param pwmChannel: The PWM channel that the Nidec Brushless controller is attached to.
+                0-9 are on-board, 10-19 are on the MXP port
+        :param dioChannel: The DIO channel that the Nidec Brushless controller is attached to.
+                0-9 are on-board, 10-25 are on the MXP port
+        """
+        self.dio = DigitalOutput(dioChannel)
+        self.dio.setPWMRate(15625)
+        self.dio.enablePWM(0.5)
+
+        self.pwm = PWM(pwmChannel)
```
Comment (Twinters007) [inline wpilib/wpilib/nidecbrushless.py]: You don't import PWM

File: wpilib/wpilib/nidecbrushless.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        self.pwm.setRaw(0xffff)
+
+        self.speed = 0.0
+        self.isInverted = False
+        self.valueEntry = None
+        self.valueListener = None
+
+        LiveWindow.addActuator("Nidec Brushless", pwmChannel, self)
+        hal.report(hal.UsageReporting.kResourceType_NidecBrushless, pwmChannel)
```
Comment (Twinters007) [inline wpilib/wpilib/nidecbrushless.py]: NidecBrushless isn't defined. Add it to the usagereporting as 62

File: wpilib/wpilib/nidecbrushless.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+        Stop the motor. This is called by the MotorSafetyHelper object
+        when it has a timeout for this PWM and needs to stop it from running.
+        """
+        self.disable()
+
+    def getDescription(self):
+        """
+        :rtype: str
+        """
+        return "Nidec " + self.getChannel()
```
Comment (auscompgeek) [inline wpilib/wpilib/nidecbrushless.py]: string plus int?

File: wpilib/wpilib/nidecbrushless.py (hunk 1 above)
Comment (virtuald) [reply wpilib/wpilib/nidecbrushless.py]: I'm ok with this.

Comment (auscompgeek) [reply wpilib/wpilib/nidecbrushless.py]: Yeah, but this is a comment in upstream, which is intended to refer to the several methods below as well.

**Comment by virtuald** [comment]
Hm? Did you mean to close this?

**Comment by ariovistus** [comment]
oh hey I can edit the title. its waiting on #331, will resubmit when that's in

**Comment by virtuald** [comment]
Looks like the tests failed, reverted the merge. Can't seem to reopen this.

**Pull request merged**